src/pages/Lending/hooks/useRepaymentLockData.tsx

```diff
@@ -0,0 +1,89 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { getThorchainLendingPosition } from 'state/slices/opportunitiesSlice/resolvers/thorchainLending/utils'
+
+type UseLendingPositionDataProps = {
+  accountId?: AccountId
+  assetId?: AssetId
+}
+
+// non-exhaustive, we only use this to get the current blockheight
+type ThorchainBlock = {
+  header: {
+    height: number
+  }
+}
+
+export const useRepaymentLockData = ({ accountId, assetId }: UseLendingPositionDataProps) => {
+  const repaymentLockQueryKey: [
+    string,
+    { accountId: AccountId | undefined; assetId: AssetId | undefined },
+  ] = useMemo(() => ['thorchainLendingRepaymentLock', { accountId, assetId }], [accountId, assetId])
+
+  const repaymentLockData = useQuery({
+    staleTime: Infinity,
+    queryKey: repaymentLockQueryKey,
+    queryFn: async () => {
+      const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
+
+      const maybePosition = await (async () => {
+        if (!(accountId && assetId)) return null
+        const position = await getThorchainLendingPosition({ accountId, assetId })
+        return position
+      })()
+
+      const maybeBlockHeight = await (async () => {
+        if (!maybePosition) return null
+        const { data: block } = await axios.get<ThorchainBlock>(`${daemonUrl}/lcd/thorchain/block`)
+        const blockHeight = block.header.height
+        return blockHeight
+      })()
+
+      const { data: mimir } = await axios.get<Record<string, unknown>>(
+        `${daemonUrl}/lcd/thorchain/mimir`,
+      )
+      // TODO(gomes): this is the repayment lock of the pool - not the borrower's
+      // we will want to make it programmatic in case there's an active position.
+      // https://dev.thorchain.org/thorchain-dev/lending/quick-start-guide
+      if ('LOANREPAYMENTMATURITY' in mimir)
+        return {
+          repaymentMaturity: mimir.LOANREPAYMENTMATURITY as number,
+          maybePosition,
+          maybeBlockHeight,
+        }
+      return null
+    },
+    select: data => {
+      if (!data) return null
+      const { repaymentMaturity, maybePosition, maybeBlockHeight } = data
+      // Current blocktime as per https://thorchain.network/stats
+      const thorchainBlockTime = '6.1'
```

Comment:
```suggestion
      const thorchainBlockTimeSeconds = '6.1'
```

Reply:
should we be fetching this rather than hardcoding it?

Reply:
Seems to be a fairly consistent ~6.1s but agree programmatic would be better, only haven't found a way to get it programmatically :(

Reply:
Looks like even THOR projects are using hardcoded values though! https://github.com/thorchain/thornode-network/blob/52187e9b98c007172b2573e3273ee7f1fe6b2dcd/src/commonFunctions.js#L19